data/core/editor/help.cfg

```diff
@@ -256,36 +304,48 @@ These following tools are provided:
 [topic]
     id=editor_palette
     title= _ "Editor Palette"
-    text= _ "The editor palette contains the applicable items you may use with the currently selected tool. For example, the Paint tool will display a full list of all available terrains, and the unit tool will provide a list of available units."
+    text= _ "The editor palette contains the applicable items you may use with the currently selected tool. For example, the Paint tool will display a full list of all available terrains, and the unit tool will provide a list of available units. When using the Starting Locations Tool, the palette changes to a list of “Player 1”, “Player 2”, etc.
+
+<bold>text='Filter'</bold>
+
+There is a filter function to show only a subset of the available items — this is the leftmost of the four buttons at the top of the palette, and the graphic changes depending on what is selected. Examples:" +
+        "<img>src=icons/terrain/terrain_group_all_30.png align=left box=yes</img>" + _ "Show all kinds of terrain" +
+        "<img>src=icons/terrain/terrain_group_water_deep_30.png align=left box=yes</img>" + _ "Show only water terrains" +
+        "<img>src=icons/terrain/terrain_group_village_30.png align=left box=yes</img>" + _ "Show only villages"
 [/topic]
 # wmllint: markcheck on
 
 # wmllint: markcheck off
+# This section uses << >> quotes so that curly brackets can be used without being interpreted by the preprocessor
+# Note: wmllint will attempt to improve lines that include, without the space character, "map_data ="
```

Comment:
What's the purpose of this comment line? Does wmllint make changes to this block that should be reverted, after running wmllint and before committing its changes?

Reply:
The version in the PR is okay, wmllint won't make changes to the file as it is now. The purpose of the comment is to stop someone thinking "there's usually no space in `map_data space =`", and removing that space.

If the `map_data =` on this comment line is changed to `map_data=`, then the file will be mangled by https://github.com/wesnoth/wesnoth/blob/704aadd17ca67dd15ba00e68d7d1861bbc789efe/data/tools/wmllint#L2649-L2654

Similarly, in the translatable string:
* `map_data=(filename in curly brackets)` caused wmllint to mangle the file
* `map_data =(filename in curly brackets)` doesn't match the regular expression, so wmllint doesn't mangle it
* `wmllint: ignore` and similar aren't useful here - it still does the mangling

When I last pushed an update I hadn't realised that the workaround was no longer needed, because  the line in the translatable string itself has literal curly brackets, it's okay both with and without that extra whitespace, because wmllint's line 2652 doesn't mangle lines with curly brackets.

What would be the clearest way to say this?

Reply:
Honestly, it sounds like we should fix wmllint to not look for `map_data` in comment lines, and maybe also to do a regexp test, `re.search("map_data *=", line)`, instead of a substring test. But that's beyond the scope of the issue.

In the meantime, how about this:

``# Note: If you change the following description string or this comment line, run wmllint on this file afterwards and make sure wmllint doesn't "improve" the map_data={...} lines.``

which uses curlies _in the comment_ to fool wmllint. But that's not cunning, that's just brittle.

We might also remove the spaces around the = sign in the translatable string. It's nice to write example code in a way that makes wmllint handle it correctly if someone copies it into a real addon.

By the way, I forgot to say, thanks for writing all the new prose here! I don't want to appear to be nitpicking, but that wmllint comment is the only thing I had a question about when I read the diff.

Reply:
Thanks.